src/client/app/effects/snippet.ts

```diff
@@ -288,6 +288,13 @@ export class SnippetEffects {
             case Snippet.ImportType.SAMPLE:
             case Snippet.ImportType.URL:
             case Snippet.ImportType.GIST:
+                let viewUrlMatch = new RegExp(`^${environment.current.config.editorUrl}\/#\/view`);
```

Comment:
I worry about taking semi-arbitrary strings and stuffing them into a regez.  What if they contain regex-special characters.  I would do a string match first instead, then trim out everything post the root URL, and regex only that.